Training/Training/file.c

```diff
@@ -1,13 +1,169 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
+// Copyright (c) Gokul TS @Metamation India.
 // ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
+// file.c
+// C Program to implement Bubble sort and binary search algorithm with test cases
 // ------------------------------------------------------------------------------------------------
-
 #include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+
+#define MAX_DIGIT 13
+#define TEST_SIZE 7
+#define ANSI_RESET_ALL "\x1b[0m"
+#define ANSI_COLOR_RED "\x1b[31m"
+#define ANSI_COLOR_GREEN "\x1b[32m"
+#define ANSI_COLOR_YELLOW "\x1b[33m"
+
+/// <summary>Changes the input array into a sorted array</summary>
+void BubbleSort (int arr[], int size);
+/// <summary>Returns the index of key element in an array</summary>
+int BinarySearch (int sortArr[], int key, int size);
+/// <summary>Gets integer input from user</summary>
+int GetUserInput (char* prompt);
+/// <summary>Implements the search and sort algorithm</summary>
+void SearchAndSort ();
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+/// <summary>Checks test cases</summary>
+void TestCases ();
+/// <summary>Compares two arrays and returns the result</summary>
+int CompareArrays (int* input, int* output, int size);
+
+int main () {
+   while (1) {
+      int num = GetUserInput ("1-SORT AND SEARCH\n2-TEST CASES\n3-CLEAR SCREEN\n4-EXIT\nEnter an option:");
```

Comment:
You can print this prompt outside the function. This function is defined to get the user input, the printing of the prompt can be done outside which can increase the flexibility of the program. If in certain case, you do not want to print any prompt, but you want to get an input you would need to pass an empty string as an argument.